opensrp-reveal/src/main/java/org/smartregister/reveal/util/NativeFormProcessor.java

```diff
@@ -0,0 +1,211 @@
+    }
+
+    private org.smartregister.domain.db.Client getDomainClient() throws JSONException {
+        if (hasClient && domainClient == null) {
+            JSONObject clientJson = new JSONObject(gson.toJson(createClient()));
+            domainClient = gson.fromJson(clientJson.toString(), org.smartregister.domain.db.Client.class);
+        }
+
+        return domainClient;
+    }
+
+    private org.smartregister.domain.db.Event getDomainEvent() throws JSONException {
+        if (domainEvent == null) {
+            JSONObject eventJson = new JSONObject(gson.toJson(createEvent()));
+            domainEvent = gson.fromJson(eventJson.toString(), org.smartregister.domain.db.Event.class);
+        }
+
+        return domainEvent;
+    }
+
+    public NativeFormProcessor saveEvent() throws JSONException {
+        JSONObject eventJson = new JSONObject(gson.toJson(createEvent()));
+
+        // inject the details
+        if (jsonForm.has(DETAILS))
+            eventJson.put(DETAILS, jsonForm.getJSONObject(DETAILS));
+
+        getSyncHelper().addEvent(createEvent().getBaseEntityId(), eventJson);
+        return this;
+    }
+
+    public NativeFormProcessor saveClient() throws JSONException {
+        JSONObject clientJson = new JSONObject(org.smartregister.family.util.JsonFormUtils.gson.toJson(createClient()));
+        getSyncHelper().addClient(createClient().getBaseEntityId(), clientJson);
+        return this;
+    }
+
+    public NativeFormProcessor clientProcessForm() throws JSONException {
+        EventClient eventClient = new EventClient(getDomainEvent(), getDomainClient());
+
+        RevealClientProcessor.getInstance(getInstance().getApplicationContext()).processClient(Collections.singletonList(eventClient), true);
+        long lastSyncTimeStamp = allSharedPreferences.fetchLastUpdatedAtDate(0);
+        Date lastSyncDate = new Date(lastSyncTimeStamp);
+        allSharedPreferences.saveLastUpdatedAtDate(lastSyncDate.getTime());
+        return this;
+    }
+
+    public NativeFormProcessor closeRegistrationID(String uniqueIDKey) throws JSONException {
+        JSONObject field = JsonFormUtils.getFieldJSONObject(JsonFormUtils.fields(jsonForm), uniqueIDKey);
+        if (field != null) {
+            String uniqueID = field.getString(JsonFormConstants.VALUE);
+            CoreLibrary.getInstance().context().getUniqueIdRepository().close(uniqueID);
+        }
+        return this;
+    }
+
+    public String getFieldValue(String fieldKey) throws JSONException {
+        String value = null;
+
+        JSONObject field = JsonFormUtils.getFieldJSONObject(JsonFormUtils.fields(jsonForm), fieldKey);
+        if (field != null && field.has(JsonFormConstants.VALUE))
+            value = field.getString(JsonFormConstants.VALUE);
+
+        return StringUtils.isBlank(value) ? "" : value;
+    }
+
+    private ECSyncHelper getSyncHelper() {
+        return ECSyncHelper.getInstance(RevealApplication.getInstance().getContext().applicationContext());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'RevealApplication.getInstance' due to existing static import 'org.smartregister.reveal.application.RevealApplication.getInstance'](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=5414138)